src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlDocletWriter.java

```diff
@@ -1369,7 +1369,9 @@ public Boolean visitInheritDoc(InheritDocTree node, Content content) {
                 @Override
                 public Boolean visitStartElement(StartElementTree node, Content content) {
                     Content attrs = new ContentBuilder();
-                    if (node.getName().toString().matches("(?i)h[1-6]")) {
+                    if (node.getName().toString().matches("(?i)h[1-6]")
+                            && !(HtmlDocletWriter.this instanceof IndexWriter)
+                            && !(HtmlDocletWriter.this instanceof SummaryListWriter<?>)) {
```

Comment:
Should we add a `Writer` method to identify primary/derived/secondary writers such as `isPrimaryWriter()` or `isDerivedWriter()`? The `instanceof` check with these two classes looks a bit fragile and untidy.

Reply:
This could be merged with a solution for [JDK-8318327](https://bugs.openjdk.org/browse/JDK-8318327) Merge Navigation.PageMode and HtmlConfiguration.ConditionalPage.

when we fix that.